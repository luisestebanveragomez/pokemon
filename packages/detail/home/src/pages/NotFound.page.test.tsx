import { fireEvent, render } from '@testing-library/react';
import NotFoundPage from './NotFound.page';

jest.mock("react-router-dom", () => ({
  __esModule: true,
  useNavigate: () => (key: string) => key,
}));

jest.mock('react-i18next', () => ({
  __esModule: true,
  Trans: ({ i18nKey }) => i18nKey,
  useTranslation: () => ({
    t: (key: string) => key,
  }),
}))

describe('Not Found Page', () => {
  it('Should Render', () => {
    const { container } = render(<NotFoundPage />)
    const button = container.querySelector("button");
    fireEvent.click(button);
    expect(container).toMatchSnapshot()
  })
})