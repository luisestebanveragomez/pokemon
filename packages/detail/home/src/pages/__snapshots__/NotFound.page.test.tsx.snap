// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found Page Should Render 1`] = `
<div>
  <section>
    <h2
      class="generic__text"
    >
      notFound.text
    </h2>
    <button
      class="generic__button"
      type="button"
    >
      notFound.back
    </button>
  </section>
</div>
`;
